src/edu/csus/ecs/pc2/ui/AutoJudgingMonitor.java

```diff
@@ -188,9 +191,13 @@ public Run findNextAutoJudgeRun() {
 
         Run[] runs = contest.getRuns();
         Arrays.sort(runs,new RunComparatorByElapsedRunIdSite());
+        
+        //make sure that any runs which were previously put on the "unavailable runs" list but whose "expiration time"
+        // for being on that list has passed get removed from the list, so that the following loop will (again) consider them.  
+        getUnavailableRunsList().removeExpiredRuns();
```

Comment:
Surround removeExpiredRuns with try/catch to avoid condition where an Exception during removeExpiredRuns 
would cause no run to be selected to be judged.

Reply:
Yes.  Agreed.